### PR TITLE
Guild method for adding bots + some eslint warn fixes

### PR DESCRIPTION
**Pull request by Vap0r1ze**

### CODE REVIEW

File: src/structures/Guild.js [hunk 1]
```diff
@@ -504,6 +504,16 @@ class Guild {
   }
 
   /**
+   * Adds a bot user to the Guild, this will not work for bot accounts.
+   * @param {string} clientID The Client ID of the bot, this isn't always the bot's User ID
+   * @param {number} [permissions=0] An optional permissions number to give the bot upon authorization
+   * @returns {Promise<Object>}
+   */
+  addBot(clientID, permissions = 0) {
```
Comment (hydrabolt) [inline src/structures/Guild.js]: Please allow permissions to also take an array of [PermissionResolvables](https://discord.js.org/#!/docs/tag/master/typedef/PermissionResolvable), it makes it easier for the end user.

File: src/client/rest/RESTMethods.js [hunk 2]
```diff
@@ -720,6 +720,19 @@ class RESTMethods {
...
+
+  authorizeBot(guildID, clientID, permissions) {
+    return new Promise((resolve, reject) => {
+      this.rest.makeRequest('post', Constants.Endpoints.oauth2Auth(clientID, permissions, true), true, {
+        guild_id: guildID,
+        permissions,
+        authorize: true,
+      })
+      .then(data => {
+        resolve(data);
```
Comment (hydrabolt) [inline src/client/rest/RESTMethods.js]: It would be nice to wrap the response in a data structure

Comment (Vap0r1ze) [inline src/client/rest/RESTMethods.js]: the endpoint doesn't give any data other than errors

File: src/structures/Guild.js (hunk 1 above)
Comment (Vap0r1ze) [inline src/structures/Guild.js]: added

File: src/structures/Guild.js [hunk 3]
```diff
@@ -504,6 +504,22 @@ class Guild {
...
 
   /**
+   * Adds a bot user to the Guild.
+   * <warn>This is only available for user accounts, not bot accounts!</warn>
+   * @param {string} clientID The Client ID of the bot, this isn't always the bot's User ID
+   * @param {PermissionResolvable[]|number} [permissions=0] An optional array of PermissionResolvables or
+   * permissions number to give the bot upon authorization
+   * @returns {Promise}
+   */
+  addBot(clientID, permissions = 0) {
```
Comment (GusCaplan) [inline src/structures/Guild.js]: change to something more representative of the actual action happening, like authenticateBot or authorizeBot something

File: src/util/Constants.js [hunk 4]
```diff
@@ -117,6 +117,15 @@ const Endpoints = exports.Endpoints = {
...
   // webhooks
   webhook: (webhookID, token) => `${API}/webhooks/${webhookID}${token ? `/${token}` : ''}`,
+
+  // oauth2
+  oauth2: `${API}/oauth2`,
+  oauth2Apps: `${Endpoints.oauth2}/applications`,
+  oauth2App: (clientID) => `${Endpoints.oauth2}/applications/${clientID}`,
+  oauth2Me: `${Endpoints.oauth2}/applications/@me`,
+  oauth2Bot: (clientID) => `${Endpoints.oauth2}/applications/${clientID}/bot`,
+  oauth2Auth: (clientID, permissions, bot) => `${Endpoints.oauth2}/authorize?client_id=${clientID}` +
```
Comment (GusCaplan) [inline src/util/Constants.js]: make this function more general to any type of auth that may be happening, maybe it could just build a url from an object

**Pull request closed without merging**